docs/extensions.adoc

```diff
@@ -0,0 +1,611 @@
+creationCallbacks: {
+    <GraphQLTypeName>: function (params) {
+        // if you want to add new fields
+        params.addFields({
+            <newFieldName>: {
+                type: <GraphQLType>,
+                args: {
+                    <argName>: <GraphQLType>,
+                    // the rest of arguments definitions ...
+                },
+            },
+        });
+
+        // if you want to remove existing fields
+        params.removeFields([
+             "fieldName 1",
+             "filedName 2",
+             // the rest of fieldName ...
+        ]);
+
+        // If you want to modify existing fields.
+        // Take aware that all arguments will be rewritten by new arguments.
+        // If type or args are not provided then will be used type and args from original field.
+        params.modifyFields({
+            <existingFieldName>: {
+                type: <GraphQLType>,
+                args: {
+                    <argName>: <GraphQLType>,
+                    // the rest of arguments definitions ...
+                },
+            },
+        });
+
+        // If you want to change description
+        params.setDescription("New description");
+
+        // If you want to add or remove some interfaces to object.
+        // Take aware that interfaces will be rewritten by new array of interfaces.
+        // Also, developer is responseble for implmentation of interface, do not forget to add all required fields.
+        params.setInterfaces([
+            <GraphQLInterfaceType or GraphQLReferenceType>
+        ]);
+    },
+    // the rest of creation callback definition ...
+}
+----
+
+For example:
+
+[source,javascript]
+----
+creationCallbacks: {
+    MyCustomType: function (params) {
+        params.addFields({
+            extraField: {
+                type: graphQL.GraphQLString,
+            }
+        });
+
+        params.removeFields([
+            "fieldToRemove1",
+            "fieldToRemove2",
+        ]);
+
+        params.modifyFields({
+          fieldToModify: {
+            type: graphQL.GraphQLString,
+          }
+        });
+    },
+
+}
+----
+
+To know how to set or override data fetcher for a field you can refer to the `resolvers` section.
+
+=== resolvers
+
+This property is used to set or override data fetcher for a field. It must be an object with the following structure:
+
+[source,javascript]
+----
+resolvers: {
+    <GraphQLTypeName>: {
+        <fieldName>: function (env) {
+            // your code here
+        },
+        // the rest of field resolvers ...
+    },
+    // the rest of type resolvers ...
+}
+----
+
+==== Application context
+
+You can assume that inside resolver function you have access to the application context where extension is defined and can use `app.name` or `app.config` properties. But in the real that will be context for Guillotine application and those properties will return `com.enonic.app.guillotine` and an actual config for Guillotine application.
```

Comment:
Most importantly - extensions.js is executed within application's JS engine and does not have direct access to other files inside application's jar file. It means that `require` function won't see neighbors files. So, when the logic is split over multiple files a bundler must be configured to produce a single file with the entire logic inlined.

Reply:
nvm. The problem is during request only. and solution is provided